programs/ssl/ssl_client2.c

```diff
@@ -3382,12 +3382,73 @@ int main( int argc, char *argv[] )
                         goto close_notify;
 
 #if defined(MBEDTLS_SSL_NEW_SESSION_TICKET)
-
                     case MBEDTLS_ERR_SSL_RECEIVED_NEW_SESSION_TICKET:
-                        /* We were waiting for application data but got a NewSessionTicket instead. */
+                        /* We received a ticket via the NewSessionTicket message.
```

Comment:
From what I see, this code-path is specific to TLS 1.3 and has been removed from TLS 1.2 code below, but tickets are also used in TLS 1.2. I would like to make sure that our work on the prototype does not make it deviate further from a dual-build with 1.2.

Could you fix this?

Reply:
The MBEDTLS_SSL_NEW_SESSION_TICKET is TLS 1.3 specific but I will double-check that I didn't make any mistake here.